src/components/dialogs/commons/grid-section.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { Grid, styled } from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import { ComponentProps, createElement, CSSProperties } from 'react';
+
+export interface GridSectionProps {
+    title: string;
+    heading?: string;
+    size?: number;
+    customStyle?: CSSProperties;
+}
+
+const createCustomTag = (heading: string, customStyle: CSSProperties) => {
+    const StyledComponent = styled('div')(() => ({
+        ...customStyle,
+    }));
+
+    return (props: ComponentProps<'div'>) => createElement(`h${heading}`, { ...props, className: StyledComponent });
+};
+
+export default function GridSection({ title, heading = '3', size = 12, customStyle }: Readonly<GridSectionProps>) {
+    const CustomTag = createCustomTag(heading, customStyle ?? {});
```

Comment:
```suggestion
export default function GridSection({ title, heading = '3', size = 12, customStyle = {} }: Readonly<GridSectionProps>) {
    const CustomTag = createCustomTag(heading, customStyle);
```

Reply:
Do not use reference type variable as literal object/array for default param in props: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-object-type-as-default-prop.md

In my opinion, the best way (as suggested by Quentin as I remember) is creating a shared emptyObj/Array in an util file then set as default value when passing the param to another component/function/hook..